### PR TITLE
Integrated yaml language server

### PR DESCRIPTION
**Pull request by JPinkney**
Signed-off-by: jpinkney <josh.pinkney@mail.utoronto.ca>

### What does this PR do?
This PR integrates the yaml language server (https://github.com/redhat-developer/yaml-language-server) into eclipse che.

pr1
pr2
pr3

### What issues does this PR fix or reference?
https://github.com/redhat-developer/rh-che/issues/213

#### Changelog

Integrated yaml language server

#### Release Notes

Integrated yaml language server

### CODE REVIEW

**Comment by riuvshin** [comment]
ci-build

File: agents/ls-yaml/src/main/resources/org.eclipse.che.ls.yaml.script.sh [hunk 1]
```diff
@@ -0,0 +1,168 @@
+#
```
[the remark this answers is not in the document]
Comment (tolusha) [reply agents/ls-yaml/src/main/resources/org.eclipse.che.ls.yaml.script.sh:1]: No shebang

File: agents/ls-yaml/src/main/resources/org.eclipse.che.ls.yaml.script.sh [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+########################
+### Install Yaml LS ###
+########################
+
+curl -s ${AGENT_BINARIES_URI} | tar xzf - -C ${LS_DIR}
+
+touch ${LS_LAUNCHER}
+chmod +x ${LS_LAUNCHER}
+echo "nodejs ${LS_DIR}/./out/server/src/server.js --stdio" > ${LS_LAUNCHER}
```
[the remark this answers is not in the document]
Comment (tolusha) [reply agents/ls-yaml/src/main/resources/org.eclipse.che.ls.yaml.script.sh]: `./out` -> `out`

File: plugins/plugin-yaml/che-plugin-yaml-ide/src/main/resources/org/eclipse/che/plugin/yaml/ide/YamlLocalizationConstant.properties [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+addUrlText=Add Schema Url
+deleteUrlText=Delete Url
+headerUiMessage=Enter a schema which can be either the url, kedge, or kubernetes. If you specify kedge or kubernetes the languageserver will specify the schema url for you.
+addUrlLabel=Add Schema url
+addSchemaButtonText=Add
+deleteUrlLabel=Delete url
+urlColumnHeader=Schema Url
+globColumnHeader=Apply To
+deleteColumnHeader=Delete
```
[the remark this answers is not in the document]
Comment (JPinkney) [reply plugins/plugin-yaml/che-plugin-yaml-ide/src/main/resources/org/eclipse/che/plugin/yaml/ide/YamlLocalizationConstant.properties]: Fixed in https://github.com/eclipse/che/pull/6371/commits/9cf7ca038147349c00b222cd26221facc63a701b. For some reason github doesn't show the correction.

File: plugins/plugin-yaml/che-plugin-yaml-ide/src/main/java/org/eclipse/che/plugin/yaml/ide/preferences/YamlExtensionManagerViewImpl.ui.xml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        <g:center>
+            <g:ScrollPanel addStyleNames="{style.emptyBorder}">
+                <g:SimplePanel>
+                    <p1:CellTable width="100%" ui:field="YamlPreferenceCellTable" focus="false"
+                                  debugId="preferences-cellTable-keys"/>
+                </g:SimplePanel>
+            </g:ScrollPanel>
+        </g:center>
+    </g:DockLayoutPanel>
+</ui:UiBinder>
```
[the remark this answers is not in the document]
Comment (JPinkney) [reply plugins/plugin-yaml/che-plugin-yaml-ide/src/main/java/org/eclipse/che/plugin/yaml/ide/preferences/YamlExtensionManagerViewImpl.ui.xml]: Fixed in https://github.com/eclipse/che/pull/6371/commits/9cf7ca038147349c00b222cd26221facc63a701b. For some reason github doesn't show the correction.

File: agents/ls-yaml/src/main/resources/org.eclipse.che.ls.yaml.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "id": "org.eclipse.che.ls.yaml",
+  "name": "Yaml language server",
+  "description": "Yaml intellisense",
+  "dependencies": [],
+  "properties": {}
+}
```
[the remark this answers is not in the document]
Comment (JPinkney) [reply agents/ls-yaml/src/main/resources/org.eclipse.che.ls.yaml.json]: Fixed in https://github.com/eclipse/che/pull/6371/commits/9cf7ca038147349c00b222cd26221facc63a701b. For some reason github doesn't show the correction.

File: plugins/plugin-yaml/che-plugin-yaml-ide/src/main/java/org/eclipse/che/plugin/yaml/ide/preferences/YamlExtensionManagerPresenter.java [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+    JsonObject parsedJson = Json.parse(jsonStr);
+    for (String glob : parsedJson.keys()) {
+      try {
+        JsonArray jsonArray = parsedJson.getArray(glob);
+        for (int arrNum = 0; arrNum < jsonArray.length(); arrNum++) {
+          YamlPreference newYamlPref = new YamlPreference(jsonArray.getString(arrNum), glob);
+          yamlPreferences.add(newYamlPref);
+        }
+      } catch (Exception e) {
+        System.out.println(e);
```
Comment (tolusha) [inline plugins/plugin-yaml/che-plugin-yaml-ide/src/main/java/org/eclipse/che/plugin/yaml/ide/preferences/YamlExtensionManagerPresenter.java]: Use logger pls.

**Review by tolusha — APPROVED** [review]

**Comment by riuvshin** [comment]
hey @JPinkney build of that PR failed on CI due to dependencies problem
```
18:08:37 [INFO] Add the following to your pom to correct the missing dependencies: 
18:08:37 [INFO] 
18:08:37 <dependency>
18:08:37   <groupId>org.slf4j</groupId>
18:08:37   <artifactId>slf4j-api</artifactId>
18:08:37   <version>1.7.24</version>
18:08:37 </dependency>
```

don't forget to sort pom after adding dep ;)

**Comment by benoitf** [comment]
@evidolob @vparfonov @skabashnyuk ?

**Review by vparfonov — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

**Comment by vparfonov** [comment]
Basically ok for me i just aware about location of binaries described here [link](https://github.com/eclipse/che/pull/6371/files#diff-46e3f3326f58fdd6611aadf65aa5ebe2R29). Is it legal @skabashnyuk ?

**Review by skabashnyuk — APPROVED** [review]

**Pull request merged**